docs/arguments.rst

```diff
@@ -34,166 +37,86 @@ And what it looks like:
 
     invoke(touch, args=['foo.txt'])
 
-Variadic Arguments
-------------------
 
-The second most common version is variadic arguments where a specific (or
-unlimited) number of arguments is accepted.  This can be controlled with
-the ``nargs`` parameter.  If it is set to ``-1``, then an unlimited number
-of arguments is accepted.
+An argument may be assigned a :ref:`parameter type <parameter-types>`. If no type is provided, the type of the default value is used. If no default value is provided, the type is assumed to be STRING.
 
-The value is then passed as a tuple.  Note that only one argument can be
-set to ``nargs=-1``, as it will eat up all arguments.
+.. admonition:: Note on Required Arguments
 
-Example:
+   It is possible to make an argument required by setting ``required=True``.  It is not recommended since we think command line tools should gracefully degrade into becoming no ops.  We think this because command line tools are often invoked with wildcard inputs and they should not error out if the wildcard is empty.
+
+Multiple Arguments
+-----------------------------------
+
+To set the number of argument use the ``nargs`` kwarg. It can be set to any positive integer and -1. Setting it to -1, makes the number of arguments arbitrary (which is called variadic) and can only be used once. The arguments are then packed as a tuple and passed to the function.
```

Comment:
I think the previous explanation was more correct and clear. This confuses "multiple arguments" and "variadic arguments". See my previous comment.

Reply:
In my view variadic is a special case of taking more than one argument, but that may not be the clearest way for documentation purposes. See above.